java/client/src/main/java/com/youtube/vitess/client/Proto.java

```diff
@@ -203,6 +210,26 @@ public EntityId apply(Map.Entry<byte[], ?> entry) {
         // Float, Double
         this.type = Query.Type.FLOAT64;
         this.value = ByteString.copyFromUtf8(value.toString());
+      } else if (value instanceof Date) {
+        // Date
+        this.type = Query.Type.DATE;
+        this.value = ByteString.copyFromUtf8(value.toString());
+      } else if (value instanceof Time) {
+        // Time
+        this.type = Query.Type.TIME;
+        this.value = ByteString.copyFromUtf8(value.toString());
+      } else if (value instanceof Timestamp) {
+        // DateTime, TimeStamp
+        this.type = Query.Type.TIMESTAMP;
+        this.value = ByteString.copyFromUtf8(value.toString());
+      } else if (value instanceof Boolean ) {
+        // Boolean
+        this.type = Query.Type.BIT;
+        this.value = ByteString.copyFromUtf8(((boolean)value) ? "1" : "0");
+      } else if (value instanceof BigDecimal) {
+        // BigDecimal
+        this.type = Query.Type.FLOAT64;
```

Comment:
I see, so you use setScale() to limit to 30. Is that really necessary? In my tests, MySQL will do the rounding for you already; it doesn't mind if the value is longer than will fit.

The reason I think we need toPlainString() instead of toString() is because toString() will revert to exponential notation if the exponent is < -6. That is, `new BigDecimal("0.0000000002").toString()` will become "2E-10". That's a problem because MySQL will interpret "2E-10" as an approximate, floating point numeric literal, rather than as an exact numeric literal:

http://dev.mysql.com/doc/refman/5.6/en/precision-math-numbers.html

To make sure MySQL interprets it as an exact numeric literal, we need to send "0.0000000002", which we can get from toPlainString().